.config/gitpod/Dockerfile

```diff
@@ -38,23 +36,37 @@ RUN PYTHON_VERSION=$(cut -d '=' -f 2 /tmp/python_version.config) && \
         vim \
         wget \
     && python3 -m pip install --no-cache-dir --upgrade pip \
-    && python3 -m pip install --no-cache-dir virtualenv \
-    # Install Docker
-    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && python3 -m pip install --no-cache-dir virtualenv
+
+# Install Docker
+FROM builder as docker-installer
+
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
     && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
     $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
     && apt-get update \
-    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io \
-    # Install npm and ungit using the versions specified in the variables
-    && curl -fsSL https://deb.nodesource.com/setup_20.x |bash - \
+    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io
+
+# Install npm and ungit
+FROM docker-installer as npm-installer
+
+RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
```

Comment:
`# hadolint ignore=DL3008`

DL3008 warning: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`: This is a best practice to ensure build reproducibility to pin the versions of installed packages. However, this can make your Dockerfile more difficult to maintain. If you want to ignore this warning, you can add `# hadolint ignore=DL3008` above the RUN lines.

Reply:
I have just made the modification, and indeed the DL3008 and DL3009 errors for line 56 do not appear during testing.

You can find the modification in this commit: beaf0c0d0c14d522e0d782a413e38c4225387500

Reply:
Hello, it would be advisable to be cautious with copy-pasting, as it can lead to significant errors. 

Even though we are discussing in the comments, it's important not to develop bad reflexes.

In this particular case, it only pertains to rule DL3008, not DL3009 as mentioned in your comment. 

Nevertheless, the commit is fine 😇